integration-tests/smoke/ccip/ccip_migration_to_v_1_6_test.go

```diff
@@ -40,6 +45,355 @@ var (
+			deployment.CreateLegacyChangeSet(v1_5.PermaBlessCommitStoreChangeset),
+			v1_5.PermaBlessCommitStoreConfig{
+				Configs: map[uint64]v1_5.PermaBlessCommitStoreConfigPerDest{
+					dest: {
+						Sources: []v1_5.PermaBlessConfigPerSourceChain{
+							{
+								SourceChainSelector: src1,
+								PermaBless:          true,
+							},
+						},
+					},
+				},
+			},
+		),
+	)
+	require.NoError(t, err)
+	oldState, err := changeset.LoadOnchainState(e.Env)
+	require.NoError(t, err)
+	require.NoError(t, err)
+	e = testhelpers.AddCCIPContractsToEnvironment(t, e.Env.AllChainSelectors(), tEnv, false)
+	// reload state after adding lanes
+
+	state, err = changeset.LoadOnchainState(e.Env)
+	require.NoError(t, err)
+	tEnv.UpdateDeployedEnvironment(e)
+
+	_, err = deployment.CreateLegacyChangeSet(v1_6.SetRMNRemoteOnRMNProxyChangeset).Apply(e.Env,
+		v1_6.SetRMNRemoteOnRMNProxyConfig{
+			ChainSelectors: e.Env.AllChainSelectors(),
+		})
+	require.NoError(t, err)
+
+	// send continuous messages in real router until done is closed
+	// send a message from the other lane src1 -> dest
+	sentEvent, err := v1_5testhelpers.SendRequest(t, e.Env, oldState,
+		testhelpers.WithSourceChain(src1),
+		testhelpers.WithDestChain(dest),
+		testhelpers.WithTestRouter(false),
+		testhelpers.WithEvm2AnyMessage(router.ClientEVM2AnyMessage{
+			Receiver:     common.LeftPadBytes(oldState.Chains[dest].Receiver.Address().Bytes(), 32),
+			Data:         []byte("hello"),
+			TokenAmounts: nil,
+			FeeToken:     common.HexToAddress("0x0"),
+			ExtraArgs:    nil,
+		}),
+	)
+	require.NoError(t, err)
+
+	_, err = deployment.CreateLegacyChangeSet(v1_6.RMNCurseChangeset).Apply(e.Env, v1_6.RMNCurseConfig{
+		CurseActions: []v1_6.CurseAction{v1_6.CurseChain(e.Env.AllChainSelectors()[0])},
+		Reason:       "Curse test",
+	})
+	require.NoError(t, err)
+
+	require.NotNil(t, sentEvent)
+	destChain := e.Env.Chains[dest]
+	require.NoError(t, err)
+	v1_5testhelpers.WaitForNoCommit(t, e.Env.Chains[src1], destChain, oldState.Chains[dest].CommitStore[src1],
+		sentEvent.Message.SequenceNumber)
+
+	commitFound := make(chan struct{})
+	go func() {
+		v1_5testhelpers.WaitForCommit(t, e.Env.Chains[src1], destChain, oldState.Chains[dest].CommitStore[src1],
+			sentEvent.Message.SequenceNumber)
+		commitFound <- struct{}{}
+	}()
+
+	_, err = deployment.CreateLegacyChangeSet(v1_6.RMNUncurseChangeset).Apply(e.Env, v1_6.RMNCurseConfig{
+		CurseActions: []v1_6.CurseAction{v1_6.CurseChain(e.Env.AllChainSelectors()[0])},
+		Reason:       "Uncurse test",
+	})
+	require.NoError(t, err)
+
+	for _, chainSel := range e.Env.AllChainSelectors() {
+		subjects, err := state.Chains[chainSel].RMNRemote.GetCursedSubjects(nil)
+		require.NoError(t, err)
+		require.Empty(t, subjects)
+	}
+
+	// We have to restart all chainlink node because it cache the curse status for 30min
+	cli, err := dockerClient.NewClientWithOpts()
+	require.NoError(t, err)
+	err = restartContainersByImage(context.Background(), cli, "chainlink")
```

Comment:
possible to use this - https://github.com/smartcontractkit/chainlink/blob/develop/integration-tests/testsetups/ccip/test_helpers.go#L130 ?